mm2src/adex_cli/src/activation_scheme_db/activation_scheme_impl.rs

```diff
@@ -0,0 +1,75 @@
+use log::{debug, error, warn};
+use serde_json::{json, Value as Json};
+use std::collections::HashMap;
+
+use super::init_activation_scheme::get_activation_scheme_path;
+use crate::helpers::read_json_file;
+
+pub struct ActivationScheme {
+    scheme: HashMap<String, Json>,
+}
+
+impl ActivationScheme {
+    fn new() -> Self {
+        Self {
+            scheme: HashMap::<String, Json>::new(),
+        }
+    }
+
+    pub fn get_activation_method(&self, coin: &str) -> Option<Json> {
+        let Some(Json::Object(object)) = self.scheme.get(coin) else { return None };
+        let mut copy = json!({});
+        for (k, v) in object.iter() {
+            // WORKAROUND: serde_json::Value does not support removing key
+            if *k == "userpass" {
+                continue;
+            }
+            copy[k] = v.clone();
+        }
+        Some(copy)
+    }
+
+    fn init(&mut self) -> Result<(), ()> {
+        let mut results: Vec<Json> = Self::load_json_file()?;
+        self.scheme = results.iter_mut().map(Self::get_coin_pair).collect();
+        Ok(())
+    }
+
+    fn get_coin_pair(element: &mut Json) -> (String, Json) {
+        let presence = element.to_string();
+        let Ok(result) = Self::get_coin_pair_impl(element) else {
+            warn!("Failed to process: {presence}");
+            return ("".to_string(), Json::Null)
+        };
```

Comment:
> -> (String, Json)
> return ("".to_string(), Json::Null)

This feels wrong to me, why not optional return?

Reply:
You're absolutely right
done